From e261ae09f000231f54340299505d00935c25f4e8 Mon Sep 17 00:00:00 2001
From: Jun Li <r65092@freescale.com>
Date: Wed, 20 May 2009 19:11:38 +0800
Subject: [PATCH] ENGR00112611 use resource_size instead of 'r->end - r->start + 1' in usb dirver.

Use resource_size() to get memory resource size instead of
'r->end - r->start + 1' in linux 2.6.28 usb driver.

Signed-off-by: Li Jun <r65092@freescale.com>
---
 drivers/usb/gadget/arcotg_udc.c |    8 ++++----
 drivers/usb/host/ehci-arc.c     |    2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/usb/gadget/arcotg_udc.c b/drivers/usb/gadget/arcotg_udc.c
index 0c393aa..d2200d8 100644
--- a/drivers/usb/gadget/arcotg_udc.c
+++ b/drivers/usb/gadget/arcotg_udc.c
@@ -2636,7 +2636,7 @@ static int __init fsl_udc_probe(struct platform_device *pdev)
 		goto err1a;
 	}
 
-	if (!request_mem_region(res->start, res->end - res->start + 1,
+	if (!request_mem_region(res->start, resource_size(res),
 				driver_name)) {
 		ERR("request mem region for %s failed \n", pdev->name);
 		ret = -EBUSY;
@@ -2645,7 +2645,7 @@ static int __init fsl_udc_probe(struct platform_device *pdev)
 #endif
 
 	if ((pdata->port_enables & FSL_USB2_DONT_REMAP) == 0) {
-		dr_regs = ioremap(res->start, res->end - res->start + 1);
+		dr_regs = ioremap(res->start, resource_size(res));
 		if (!dr_regs) {
 			ret = -ENOMEM;
 			goto err1;
@@ -2786,7 +2786,7 @@ err2a:
 		iounmap(dr_regs);
 err1:
 	if (!udc_controller->transceiver)
-		release_mem_region(res->start, res->end - res->start + 1);
+		release_mem_region(res->start, resource_size(res));
 err1a:
 	kfree(udc_controller);
 	udc_controller = NULL;
@@ -2826,7 +2826,7 @@ static int __exit fsl_udc_remove(struct platform_device *pdev)
 
 #ifndef CONFIG_USB_OTG
 	res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
-	release_mem_region(res->start, res->end - res->start + 1);
+	release_mem_region(res->start, resource_size(res));
 #endif
 
 	device_unregister(&udc_controller->gadget.dev);
diff --git a/drivers/usb/host/ehci-arc.c b/drivers/usb/host/ehci-arc.c
index 4c17193..009001c 100644
--- a/drivers/usb/host/ehci-arc.c
+++ b/drivers/usb/host/ehci-arc.c
@@ -170,7 +170,7 @@ static int usb_hcd_fsl_probe(const struct hc_driver *driver,
 
 		res = platform_get_resource(pdev, IORESOURCE_MEM, 0);
 		hcd->rsrc_start = res->start;
-		hcd->rsrc_len = res->end - res->start + 1;
+		hcd->rsrc_len = resource_size(res);
 
 		if (!request_mem_region(hcd->rsrc_start, hcd->rsrc_len,
 					driver->description)) {
-- 
1.5.4.4

